tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,259 @@
+
+OBSERVED_DENOMINATORS = [100_000, 1_000_000, 10_000_000]
+TARGET_PROPORTION = 0.1
+BOUNDS = [
+    1e-14,
+    1e-10,
+    0.000001,
+    0.01,
+    0.1,
+    0.2,
+    0.3,
+    0.4,
+    0.5,
+    0.6,
+    0.8,
+    0.9,
+    0.99999,
+    1.0 - 1e-10,
+]
+WIDTHS = [
+    1e-14,
+    1e-12,
+    1e-10,
+    0.0000001,
+    0.00001,
+    0.001,
+    0.01,
+    0.03,
+    0.05,
+    0.1,
+    0.2,
+    0.4,
+    0.6,
+    0.9,
+]
+
+
+@pytest.mark.parametrize(
+    "numerator, denominator, target_proportion",
+    [(10_008, 100_000, 0.1), (976, 1_000_000, 0.001), (1_049, 50_000, (0.0198, 0.0202))],
+)
+def test_pass_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float
+) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+@pytest.mark.parametrize(
+    "numerator, denominator, target_proportion, match",
+    [
+        (901, 100_000, 0.05, "is significantly less than expected"),
+        (1_150, 50_000, 0.02, "is significantly greater than expected"),
+    ],
+)
+def test_fail_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float, match: str
+) -> None:
+    with pytest.raises(AssertionError, match=match):
+        FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+def test_small_sample_size_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(1, 10, 0.1)
+    assert "Sample size too small" in caplog.text
+
+
+def test_not_conclusive_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    # This test verifies we will pass, then be inconclusive, then fail
+    # The numbers used in this test are arbitrary but are intended to be conservative
+    # estimates of the number of iterations needed to reach each state
+    # This is to cache some of the computation for the while loop
+    fuzzy_checker = FuzzyChecker()
+    i = 1_000
+    while True:
+        caplog.clear()
+        fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+        if "is not conclusive" in caplog.text:
+            assert i > 1050
+            break
+        if i > 1_200:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+    while True:
+        caplog.clear()
+        try:
+            fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+            assert "is not conclusive" in caplog.text
+        except AssertionError as e:
+            assert "is significantly greater" in str(e)
+            assert i > 1_100
+            break
+        if i > 1_300:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+
+@pytest.mark.parametrize(
+    "numerator",
+    [
+        1_008,
+        10_008,
+        100_008,
+        1_000_008,
+    ],
+)
+@pytest.mark.parametrize("denominator", OBSERVED_DENOMINATORS, ids=lambda x: x)
```

Comment:
Do you need the lambda here?

Reply:
I don't think it changes the test id at all, so can be removed.